### ISSUE TITLE
IP address disappears from status line

### ISSUE THREAD

**Issue by coreice** [0xAF/tmux-public-ip#2]
I am experiencing a rather strange issue with the plugin. After installing and setting it up in the status line, the IP address appears... and disappears in the blink of an eye. See the following GIF (look at what happens before the text `ip test`).

![ezgif-2-687ddd701c](https://github.com/0xAF/tmux-public-ip/assets/10442497/515a9adc-29db-48d6-9490-26913d6c2f8e)

The only way to let the address reappear is by changing the Public IP temp file.

I am using this on macOS 13.3.1a in iTerm 3.4.19 with zsh 5.9 and Tmux 3.3a.

```tmux
set -g @public_ip_dir '~/.config/tmux/tmp/'
set -g @public_ip_tmp_file 'public_ip.json'
```

```tmux
set -g status-right "#[fg=#586e75,bg=#002b36,nobold,nounderscore,noitalics]ÓÇ≤#[fg=#93a1a1,bg=#586e75]#[fg=#eee8d5,bg=#586e75,nobold,nounderscore,noitalics] #{public_ip ip} ÔÇ¨ ip test #[fg=#657b83]ÓÇ≤#[fg=#eee8da,bg=#657b83] #{weather} #[fg=#93a1a1,bg=#657b83,nobold,nounderscore,noitalics]ÓÇ≤#[fg=#15161e,bg=#93a1a1,bold] #h ÓÇ≥ %H:%M:%S‚†Ä#[fg=#93a1a1,bg=#0d1017]ÓÇ¥‚†Ä"
```

I also tried different configuration variants like the one in the readme but with no luck.

Any ideas what is going wrong here?

Update: I also notice the JSON file does not get updated after initially creating, but it does get recreated after removing it.

**Comment by coreice** [comment]
After some debugging I found out the issue lies in the `stat` command used in the statement below: 

https://github.com/0xAF/tmux-public-ip/blob/871a2e1e9904dce758cb43c38acad38810c91e7b/scripts/public_ip.sh#L64

Running this command separately in bash (on macOS) gave me this error (after setting the `$cache` variable of course):
```
bash-3.2$ stat -L --format %Y "$cache"
stat: illegal option -- -
usage: stat [-FLnq] [-f format | -l | -r | -s | -x] [-t timefmt] [file ...]
```

I saw [this](https://unix.stackexchange.com/a/102698) post on StackExchange explaining that macOS / OSX and other unix (like) systems handle the `date` and `stat` command differently. I replaced the `date` and `stat` command in line 64 with the one from the StackExchange post and the plugin works fine now 😄.

To be complete, this is the working `if` statement for me:

```sh
if [[ ! -f "$cache" || $(($(date +%s) - $(stat -t %s -f %m -- "$cache"))) -gt $refresh ]]; then
    curl -s "$_PIP_URL" -o "$cache"
fi
```

Not sure how a good fix for this would look like (my knowledge of `sh` is limited). A check like `if Darwin { somethings }` would do it probably but I don't know if there is a more nicely solution to this.
